src/toucan/db.clj

```diff
@@ -281,15 +313,32 @@
         (maybe-qualify model field)))
     (models/default-fields (resolve-model model))))
 
+(defn- replace-underscore [^Keyword k]
+  (let [k-str (name k)]
+    (if (string/index-of k-str \_)
+      (keyword (string/replace k-str \_ \-))
+      k)))
+
+(defn- transform-keys [f m]
+  (walk/postwalk
+    (fn [x]
+      (if (map? x)
+        (into {} (map
+                   (fn [[k v]]
+                     [(f k) v])
+                   x))
+        x))
+    m))
 
 (defn do-post-select
   "Perform post-processing for objects fetched from the DB.
    Convert results OBJECTS to ENTITY record types and call the model's `post-select` method on them."
   {:style/indent 1}
   [model objects]
-  (let [model (resolve-model model)]
+  (let [model (resolve-model model)
+        post-select (if (allow-dashed-names) identity (partial transform-keys replace-underscore))]
```

Comment:
Added conversion of field names to replace underscore with dash.
@camsaul  should the config option instead by called `underscored-field-names` (which defaults to `false`)?